server/Arcadia.Assistant.Calendar/Vacations/EmployeeVacationsActor.cs

```diff
@@ -168,11 +133,32 @@ protected override void UpdateCalendarEvent(CalendarEvent oldEvent, CalendarEven
             onUpsert(newEvent);
         }
 
+        protected override void ApproveCalendarEvent(ApproveCalendarEvent message, OnSuccessfulApproveCallback onSuccessfulApprove)
+        {
+            var @event = new UserGrantedCalendarEventApproval
+            {
+                EventId = message.Event.EventId,
+                TimeStamp = DateTimeOffset.Now,
+                ApproverId = message.ApproverId
+            };
+
+            this.Persist(@event, ev =>
+            {
+                this.OnUserGrantedVacationApproval(ev);
```

Comment:
implementations are the same, only difference is method called. Maybe better not override this but have that call overwritten instead?

Reply:
I thought about it. Made another way to not have Persist method call in base class. Easily can be done as you suggest.

Reply:
Done.